test/stockpile_test.clj

```diff
@@ -0,0 +1,306 @@
+(ns stockpile-test
+  (:require [stockpile :as stock]
+            [clojure.java.io :as io]
+            [clojure.java.shell :as shell]
+            [clojure.test :refer :all])
+  (:import [org.apache.commons.lang3 RandomStringUtils]
+           [java.io ByteArrayInputStream File FileNotFoundException]
+           [java.nio.file Files NoSuchFileException]
+           [java.nio.file.attribute FileAttribute]
+           [java.util.concurrent.atomic AtomicLong]))
+
+;; FIXME: add full fs tests
```

Comment:
Yeah, saw I'd overlooked that one too: https://tickets.puppetlabs.com/browse/PDB-2788